### DISCUSSION TITLE
Is there any similar to github.com/rivo/tview

### DISCUSSION THREAD

**Discussion by jorgeluismireles** [vlang/v#21933] [Questions and Answers] [answered]
I have used the **Rich Interactive Widgets for Terminal UIs** go library [tview](https://github.com/rivo/tview)
Is very helpful to add complex configuration for console applications. I check the `term.ui` code but not find *widgets*.

Thanks!

**Comment by JalonSolov** [comment, marked as answer]
V does not currently have terminal widgets.

**Comment by medvednikov** [comment]
@jorgeluismireles watch the development of https://github.com/vlang/go2v

Soon it'll be possible to translate that entire project to V.

**Reply by jorgeluismireles** [reply]
Let me try to understand. Lets say we use the go **tview** library like this:
```
	return tview.NewFlex().
		SetDirection(tview.FlexRow).
		AddItem(list, 14, 1, false).
		AddItem(tview.NewForm().
			AddButton("Save", func(){
				g.ModalYesNo("Save?", tcell.ColorGreen, tcell.ColorBlack, func() {
					if err := c.Save(g.file); err != nil {
						g.Modal("", err)
					}
				})
			}).
			AddButton("Quit", g.Quit), 0, 1, false)

```
Then, once "transpiled" one could use v language with something like this? (I am newbie in v):
```
return tview.new_flex().
   .set_direction(tview.FlexRow).
   .add_time(list, 14, 1, false).
   .add_item(tview.new_form() ...
```
Best regards.

**Reply by medvednikov** [reply]
Yes, V and Go are very similar.

**Reply by medvednikov** [reply]
` .set_direction(.flex_row)` V has enums :)
